### PR TITLE
feat(other): add browser region override support for Hong Kong and Si…

### PR DESCRIPTION
**Pull request by katie-bobbe-genesys**
…ngapore regions

### CODE REVIEW

File: packages/genesys-spark-components/src/genesys-spark-utils/intl.ts [hunk 1]
```diff
@@ -84,11 +84,23 @@ export function determineDisplayLocale(
...
+    browserLocaleOverride(localeString) ||
+    (localeString.length == 2 &&
+      navigator.language?.startsWith(`${localeString}-`))
   );
 }
 
+function browserLocaleOverride(localeString: string): boolean {
+  switch (localeString.toLowerCase()) {
+    case 'zh-cn':
+      return !!navigator.language?.toLowerCase().startsWith('zh-sg');
```
Comment (gavin-everett-genesys) [inline packages/genesys-spark-components/src/genesys-spark-utils/intl.ts:96]: Would  `return  navigator.language?.toLowerCase().startsWith('zh-sg') || false` work also ? I always find it hard to parse the double negation pattern 😆

Comment (daragh-king-genesys) [reply packages/genesys-spark-components/src/genesys-spark-utils/intl.ts:96]: `Boolean(navigator.language?.toLowerCase().startsWith('zh-hk'))` is also an option :)

**Comment by katie-bobbe-genesys** [comment]
I made a separate ticket for general utils refactors here COMUI-3087
